pep-0458-tuf-online-keys.rst

```diff
@@ -395,6 +395,208 @@ specifying `delegations`__.
+}
+
+TARGETS is an object whose format is the following:
+
+{
+  TARGETPATH : {
+    "length" : LENGTH,
+    "hashes" : HASHES,
+    ("custom" : { ... }) },
+  ...
+}
+
+Each key of the TARGETS object is a TARGETPATH.  A TARGETPATH is a path to
+a file that is relative to a mirror's base URL of targets.
+
+It is allowed to have a TARGETS object with no TARGETPATH elements.  This can
+be used to indicate that no target files are available.
+
+The HASH and LENGTH are the hash and length of the target file. If defined, the
+elements and values of "custom" will be made available to the client
+application.  The information in "custom" is opaque to the framework and can
+include version numbers, dependencies, requirements, and any other data that
+the application wants to include to describe the file at TARGETPATH.  The
+application may use this information to guide download decisions.
+
+DELEGATIONS is an object whose format is the following:
+
+{
+  "keys" : {
+    KEYID : KEY,
+    ...
+  },
+  "roles" : [{
+    "name": ROLE,
+    "keyids" : [ KEYID, ... ] ,
+    "threshold" : THRESHOLD,
+    ("path_hash_prefixes" : [ HEX_DIGEST, ... ] |
+    "paths" : [ PATHPATTERN, ... ])},
+  ...
+  ]
+}
+
+In order to discuss target paths, a role MUST specify only one of the
+"path_hash_prefixes" or "paths" attributes, each of which we discuss next.
+
+The "path_hash_prefixes" list is used to succinctly describe a set of target
+paths. Specifically, each HEX_DIGEST in "path_hash_prefixes" describes a set of
+target paths; therefore, "path_hash_prefixes" is the union over each prefix of
+its set of target paths. The target paths must meet this condition: each target
+path, when hashed with the SHA-256 hash function to produce a 64-byte
+hexadecimal digest (HEX_DIGEST), must share the same prefix as one of the
+prefixes in "path_hash_prefixes". This is useful to split a large number of
+targets into separate bins identified by consistent hashing.
+
+The "paths" list describes paths that the role is trusted to provide.  Clients
+MUST check that a target is in one of the trusted paths of all roles in a
+delegation chain, not just in a trusted path of the role that describes the
+target file.  The format of a PATHPATTERN may be either a path to a single
+file, or a path to a directory to indicate all files and/or subdirectories
+under that directory.
+
+
+timestamp.JSON
+~~~~~~~~~~~~~~
+
+The timestamp file is signed by a timestamp key.  It indicates the
+latest versions of other files and is frequently resigned to limit the
+amount of time a client can be kept unaware of interference with obtaining
+updates.
+
+Timestamp files will potentially be downloaded very frequently.  Unnecessary
+information in them will be avoided.
+
+The format of the timestamp file is as follows:
+
+{
+  "_type" : "Timestamp",
+  "version" : VERSION,
+  "expires" : EXPIRES,
+  "meta" : METAFILES
+}
+
+METAFILES is the same is described for the snapshot.json file.  In the case of
```

Comment:
595: "metafiles is the same" ____ (what?) then, use "is" or "as" depending on what works